symfony/stimulus-bundle/2.20/assets/controllers/csrf_protection_controller.js

```diff
@@ -1,60 +1,78 @@
-var nameCheck = /^[-_a-zA-Z0-9]{4,22}$/;
-var tokenCheck = /^[-_/+a-zA-Z0-9]{24,}$/;
+const nameCheck = /^[-_a-zA-Z0-9]{4,22}$/;
+const tokenCheck = /^[-_/+a-zA-Z0-9]{24,}$/;
 
 // Generate and double-submit a CSRF token in a form field and a cookie, as defined by Symfony's SameOriginCsrfTokenManager
 document.addEventListener('submit', function (event) {
-    var csrfField = event.target.querySelector('input[data-controller="csrf-protection"]');
+    generateCsrfToken(event.target);
+}, true);
+
+// When @hotwired/turbo handles form submissions, send the CSRF token in a header in addition to a cookie
+// The `framework.csrf_protection.check_header` config option needs to be enabled for the header to be checked
+document.addEventListener('turbo:submit-start', function (event) {
+    Object.entries(generateCsrfHeaders(event.detail.formSubmission.formElement)).forEach(([name, value]) => {
```

Comment:
Object.entries isn't available with IE11, while I think the previous script was working with it.
`export`  is also a feature not compatible with IE11 so that maybe the script didn't work with it already, but maybe [the eshim](https://github.com/guybedford/es-module-shims) made in work in the end?

What I mean is: if the snippet worked with IE11, let's not break IE11 support. Any idea of the status of this?

Reply:
Turbo and Stimulus do not support IE 11, so I believe it is fine to use more modern API's

Reply:
No IE support for Stimulus, Turbo, import map, nor the importmap polyfill  (see [here](https://github.com/guybedford/es-module-shims/issues/14#issuecomment-493261534))

No consequence / difference with today ☺️

Reply:
this comment is old, the polyfill does support IE11 (at least the .dist version, I've tried already)

Reply:
The `dist` file of `es-module-shims` uses `async`/`await` among other things. So it definitely does not support IE11 without transpiling. And it also uses some modern methods that won't work without polyfiling them.

Projects needing to support IE11 probably have an infrastructure to transpile code to ES5 and load polyfills, which would work for this file as well.